Assistant/update machine generated text

rosannamilner opened the pull request:
Machine Generated Text service now provides sentence scores allowing for sentence highlighting like other credibility signals

jmafoster1 commented:
This mostly looks good. I noticed that the tails of letters like p and g overflowed the bottom of the highlighting slightly, so I fixed that with a small amount of padding. A couple of other points:
- I noticed, when I scraped [sheffield.ac.uk](sheffield.ac.uk), that the "PhD study" heading seems to be duplicated. I'm not sure why, perhaps it's right on a threshold between "likely human written" and "highly likely human written" and there's a control logic bug that shows both?
![image](https://github.com/user-attachments/assets/d9bd0c84-57a5-4f29-9f10-a1b18eb02ea4)
- On the orange highlighted text, the white text is a little difficult to read. I think that text might be easier to read if it was black.
- The machine generated text classifier can be very slow to run. I refreshed the page a couple of times, and was sat there for about a minute after everything else had loaded, then the request timed out. I refreshed the page again and it took a few seconds, but was definitely still after everything else had already loaded.

Sallaa commented:
Thank you for the PR @rosannamilner, I am going to ask Denis if I can get the green light to merge this.

The only comment I have is that you have a few places where you commented some old code, maybe you can clean it up and delete these comments instead?

Other than this it looks good to me! 🎉

rosannamilner commented:
Thanks both for the comments - I will look into the points (and I've just spotted at error with the DBKF media results) so I'll message when these are sorted.

rosannamilner commented:
> This mostly looks good. I noticed that the tails of letters like p and g overflowed the bottom of the highlighting slightly, so I fixed that with a small amount of padding. A couple of other points:
> 

Thanks for fixing the padding!

> * I noticed, when I scraped [sheffield.ac.uk](sheffield.ac.uk), that the "PhD study" heading seems to be duplicated. I'm not sure why, perhaps it's right on a threshold between "likely human written" and "highly likely human written" and there's a control logic bug that shows both?
>   ![image](https://private-user-images.githubusercontent.com/13611658/430845636-d9bd0c84-57a5-4f29-9f10-a1b18eb02ea4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMjEyOTYsIm5iZiI6MTc0NDAyMDk5NiwicGF0aCI6Ii8xMzYxMTY1OC80MzA4NDU2MzYtZDliZDBjODQtNTdhNS00ZjI5LTlmMTAtYTFiMThlYjAyZWE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDEwMTYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRmNDJlOGY2N2QxZjY4ZjQ3NzBmMjY1NjQ1M2Q1YzQxNTllMjAwODczZmEwNDJjMDU5OTVmNzIwMjU3OGE4ZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6OBYdnlF4zpc-KnsxsJy-BYsCgDWRxNZy7ts-BrVvtA)
>   If you look on the actual website, it's definitely only there once, even in the source code.

This is an issue with the returned results so I'll go back to Ivan about this. It's returned an overlapping segment.

>   ![image](https://private-user-images.githubusercontent.com/13611658/430847870-8879224e-40c6-48e7-88e4-8c9a07c2cf6a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMjEyOTYsIm5iZiI6MTc0NDAyMDk5NiwicGF0aCI6Ii8xMzYxMTY1OC80MzA4NDc4NzAtODg3OTIyNGUtNDBjNi00OGU3LTg4ZTQtOGM5YTA3YzJjZjZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDEwMTYzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4NTFlZGNhMGIwODE4YTgyYjAxMTYyNzYxYzZlN2EzODQ3YzY1ZWFkNmE5MjFmMGVhNjcxMzA1YzYyN2I3MDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wvLuMADPlGjGfhgCtbNhwkgM00XhFHOjuPa-CxZbGCc)
> * On the orange highlighted text, the white text is a little difficult to read. I think that text might be easier to read if it was black.

I have changed this to black. It's decided automatically using `rgbToLuminance()` in `assistantUtils.jsx` but I agree in this case the black on orange looks much better.

> * The machine generated text classifier can be very slow to run. I refreshed the page a couple of times, and was sat there for about a minute after everything else had loaded, then the request timed out. I refreshed the page again and it took a few seconds, but was definitely still after everything else had already loaded.

Yes they've made me aware about this in the past, I can mention again in case there's a way they can speed it up.

rosannamilner commented:
> Thank you for the PR @rosannamilner, I am going to ask Denis if I can get the green light to merge this.
> 
> The only comment I have is that you have a few places where you commented some old code, maybe you can clean it up and delete these comments instead?
> 
> Other than this it looks good to me! 🎉

I've removed all the commented old code that I can see, let me know if I've missed any!

I'm happy with this to be merged. I'm waiting for a response from kinit about the overlapping segment Michael spotted, but I believe this is an issue at their end.

jmafoster1 commented:
@Sallaa as requested, I've resolved the conflicts. (Also worth noting that yesterday wasn't a working day in the UK, which may explain why Rosanna was not contactable).

Sallaa commented:
@jmafoster1 Ah, I was not aware of that. Thank you for letting me know. I will review everything and test it this afternoon.

Sallaa commented:
We have noticed a few non-critical problems, which we can take care of later, but I'm going ahead and I am merging this because partners want to access this feature asap
